cucumber/api/features/resource_visibility.feature

```diff
@@ -1,7 +1,109 @@
+
+  Scenario: Resources without permissions or ownership are not visible
+    Given I create a new resource called "probe"
+    And I create a new user "alice"
+
+    When I login as "alice"
     And I successfully GET "/resources/cucumber"
+
     Then the resource list should not include the newest resource
+
+  Scenario: Resources with permissions are visible
+    Given I create a new resource called "probe"
+    And I create a new user "alice"
+    And I permit user "alice" to "try" it
+
+    When I login as "alice"
+    And I successfully GET "/resources/cucumber"
+
+    Then the resource list should include the newest resource
+
+  Scenario: Resources with transitive permissions are visible
+
+    Note: A role has a "transitive permission" on a resource if it's a member of a role that has a permission.
+
+    Given I create a new resource called "probe"
+    And I create a new user "alice"
+    And I permit user "alice" to "try" it
+
+    And I create a new user "bob"
+    And I grant user "alice" to user "bob"
+
+    When I login as "bob"
+    And I successfully GET "/resources/cucumber"
+
+    Then the resource list should include the newest resource
+
+  Scenario: Owned resources are visible even without explicit permissions
+    Given I create a new user "alice"
+    And I login as "alice"
+    And I create a new resource called "probe"
+
+    When I successfully GET "/resources/cucumber"
+    Then the resource list should include the newest resource
+
+  Scenario: Transitively owned resources are visible even without explicit permissions
+
+    Note: a resource is "transitively owned" if the user holds a role that is the owner.
+
+    Given I create a new user "alice"
+    And I create a new user "bob"
+    And I grant user "alice" to user "bob"
+
+    And I login as "alice"
+    And I create a new resource called "probe"
+
+    When I login as "bob"
+    Then I can GET "/resources/cucumber/test-resource/probe"
+    And I can GET "/resources/cucumber"
+    And the resource list should include the newest resource
+
+  Scenario: Showing a resource without permissions
+    If the user doesn't have a permission or ownership of a resource, showing
+    that resource should return 404.
+
+    Given I create a new resource called "probe"
+    And I create a new user "alice"
+
+    When I login as "alice"
+    And I GET "/resources/cucumber/test-resource/probe"
+
+    Then the HTTP response status code is 404
+
+  Scenario: Fetching a secret without any permission on it
+    If the user doesn't have any permission or ownership of a secret, fetching
+    it should return 404 (not 403) even if it exists.
+
+    Given I create a new resource called "probe"
+    And I successfully POST "/secrets/cucumber/test-resource/probe" with body:
+    """
+    v-1
+    """
+    And I create a new user "alice"
+
+    When I login as "alice"
+    And I GET "/secrets/cucumber/test-resource/probe"
+    Then the HTTP response status code is 404
```

Comment:
Wouldn't this already be covered by the user not having `execute` on the variable?

Reply:
I think this is different. SecretsController starts by looking for the secret's resource, which will fail if the user has no privilege on the secret.

It seems like we should have scenarios analogous to this one for the other controllers that do this kind of checking (PoliciesController and HostFactoryTokensController). I'll add those.